airflow/jobs.py

```diff
@@ -2540,9 +2542,25 @@ def _execute(self, session=None):
                         self.dag_id
                     )
                     time.sleep(self.delay_on_limit_secs)
+
+        except (KeyboardInterrupt, SystemExit):
+            for dag_run in self.dag_runs:
+                dag_run.refresh_from_db(session)
+                make_transient(dag_run)
+                unfinished_tasks = dag_run.get_task_instances(
```

Comment:
I think we can first check the whether the DagRun status is SUCCESS,
If SUCCESS, we can just continue

Reply:
the dagrun status won't set it here, but set in Basejob(outside loop) in the normal state. But if it encounters ctrl^c, it will not continue the BaseJob remaining function.

Reply:
I see.